govcd/vm_test.go

```diff
@@ -986,23 +1015,80 @@ func (vcd *TestVCD) Test_GetVirtualHardwareSection(check *C) {
 	vm, err := vcd.client.Client.GetVMByHref(existingVm.HREF)
 	check.Assert(err, IsNil)
 
-	// Preform check of virtual hardware section
-	section, err := vm.GetVirtualHardwareSection()
+	storageProfile, err := vcd.vdc.FindStorageProfileReference(vcd.config.VCD.StorageProfile.SP1)
 	check.Assert(err, IsNil)
+	isThinProvisioned := true
+	iops := int64(0)
+	diskSettings := &types.DiskSettings{
+		SizeMb:            1024,
+		UnitNumber:        0,
+		BusNumber:         busNumber,
+		AdapterType:       "4",
+		ThinProvisioned:   &isThinProvisioned,
+		StorageProfile:    &storageProfile,
+		OverrideVmDefault: true,
+		Iops:              &iops,
+	}
 
-	// Check that section.Info is not Nil, as its the only field that may not be omitted when marshalled
-	check.Assert(section.Info, NotNil)
+	// disables fast provisioning if needed
+	previousVdcFastProvisioningValue := updateVdcFastProvisioning(vcd, check, "disable")
 
-	// Check that section.Item is not Nil before looping over it
-	check.Assert(section.Item, NotNil)
+	diskId, err := vm.AddInternalDisk(diskSettings)
+	check.Assert(err, IsNil)
+	check.Assert(diskId, NotNil)
+	return vm, storageProfile, diskSettings, diskId, previousVdcFastProvisioningValue, err
+}
 
-	// Loop over the Items to ensure
-	for _, item := range section.Item {
-		check.Assert(item.ResourceType, NotNil)
-		check.Assert(item.ResourceSubType, NotNil)
+// updateVdcFastProvisioning Enables or Disables fast provisioning if needed
+func updateVdcFastProvisioning(vcd *TestVCD, check *C, enable string) string {
```

Comment:
We are returning only two possible values.
Seems a task that is suited for a `bool` instead of a string.

Reply:
Yes, it's true - only two possible values. I made string that it would be more readable. updateVdcFastProvisioning(.., `enable`) vs updateVdcFastProvisioning(.., true)